### ISSUE TITLE
NotificationResponse throw Bad response format

### ISSUE THREAD

**Issue by jamesblasco** [open-rpc/client-js#194]
I think in this past issue #230 you define that notification doesn't require the result param, but in the code, it throws an error while parsing a response.

Here in [this code](https://github.com/open-rpc/client-js/blob/ce8d06f05230ab9c1ab1b31fdf3c556cc96f4938/src/transports/TransportRequestManager.ts#L54), the method `checkJSONRPC` returns false for the notification case, as it neither has an error nor result param

```ts
 public resolveResponse(payload: string, emitError: boolean = true): TransportResponse {
    let data: any = payload;
    try {
      data = JSON.parse(payload);
      if (this.checkJSONRPC(data) === false) {
        throw new Error("Bad response format");
      }
....
    return this.resolveRes(data, emitError);
....
  }

  private checkJSONRPC(data: any) {
    let payload = [data];
    if (data instanceof Array) {
      payload = data;
    }
    return payload.every((datum) => (datum.result !== undefined || datum.error !== undefined));
  }
```

Later the method `resolveRes` is the one that checks if there is not id or error and parses it as a notification response

```ts
 if (id === undefined && error === undefined) {
      this.transportEventChannel.emit("notification", data as IJSONRPCNotificationResponse);
      return;
   }

```

**Comment by jamesblasco** [comment]
Okey I just saw your PR #193, and that should fix this

**Comment by shanejonas** [comment]
@jamesblasco shipped in `1.3.3`, let us know if there are any issues. Thanks for the bug reports, super helpful.
